docs/src/main/tut/docs/stack/README.md

```diff
@@ -0,0 +1,260 @@
+
+In this example a company selling multiple types (cultivars) of apples wants to process the orders of its customers. The order needs to be validated,
+we need to check if the requested amount of apples is in stock and in the end register the order.
+
+We start with the imports of the Freestyle core.
+
+```tut:book
+import freestyle._
+import freestyle.implicits._
+```
+
+Now we can create some data types for a customer and an order together with a `Config` data type which holds a set of all the apple cultivars the company
+sells.
+
+```tut:book
+import java.util.UUID
+
+type CustomerId = UUID
+
+case class Customer(id: CustomerId, name: String)
+case class Order(crates: Int, cultivar: String, customerId: CustomerId)
+
+case class Config(cultivars: Set[String])
+```
+
+It should be possible to get a `Customer` for a specified `CustomerId` from some data store holding the company's customers.
+We should be able to check how much crates of some specific apple cultivar the company has in stock and finally to register an order.
+
+We will represent these capabilities using two algebras: `CustomerPersistence` and `StockPersistence`.
+
+```tut:book
+object algebras {
+  @free trait CustomerPersistence[F[_]] {
+    def getCustomer(id: CustomerId): FreeS.Par[F, Option[Customer]]
+  }
+
+  @free trait StockPersistence[F[_]] {
+    def checkQuantityAvailable(cultivar: String): FreeS.Par[F, Int]
+    def registerOrder(order: Order): FreeS.Par[F, Unit]
+  }
+}
+```
+
+These two algebras will be combined together in a `Persistence` module, that in its turn will be a part of our `App` module, that additionally includes the
+capabilities to fail, to read from our `Config` configuration and to accumulate some logging messages using respectively the `error`, the `reader` and  the `writer` effect.
+
+```tut:book
+import freestyle.effects.error._
+import freestyle.effects.{reader, writer}
+
+object modules {
+  val rd = reader[Config]
+  val wr = writer[Vector[String]]
+
+  @module trait Persistence[F[_]] {
+    val customer: algebras.CustomerPersistence[F]
+    val stock: algebras.StockPersistence[F]
+  }
+
+  @module trait App[F[_]] {
+    val persistence: Persistence[F]
+
+    val errorM: ErrorM[F]
+    val readerM: rd.ReaderM[F]
+    val writerM: wr.WriterM[F]
+  }
+}
+```
+
+To validate the order we check if the number of crates that is ordered bigger is than zero and that the requested apple cultivar is selled by the company.
+
+For the second part we use the `reader` effect algebra to read the set of apple cultivars from our `Config`.
+
+```tut:book
+import cats.implicits._
+
+import modules._
+import cats.data.{NonEmptyList, ValidatedNel}
+
+def validateOrder[F[_]](order: Order, customer: Customer)(implicit app: App[F]): FreeS.Par[F, ValidatedNel[String, Unit]] =
+  app.readerM.reader { config =>
+    val v = ().validNel[String]
+    v.ensure(NonEmptyList.of(
+      "Number of crates ordered should be bigger than zero"))(
+      _ => order.crates > 0) |+|
```

Comment:
Since new users may not be familair with `|+|` we should add some text in the description of this paragraph that briefly explain that it comes from cats and what it's purpose is.